lib/daru/maths/statistics/vector.rb

```diff
@@ -377,6 +377,36 @@ def sample_without_replacement(sample=1)
           end
         end
 
+        # The percent_change method computes the percent change over
+        # the given number of periods.
+        #
+        # @param [Integer] periods (1) number of nils to insert at the beginning.
+        #
+        # @example
+        # 
+        #   vector = Daru::Vector.new([6,4,2,nil,'t',1], index: ['a','b','c','d','e','f'])
+        #   vector.percent_change
+        #   #=>
+        #   #   <Daru::Vector:28713060 @name = nil @size: 4 >
+        #   #              nil
+        #   #   a	
+        #   #   b	   -0.3333333333333333
+        #   #   c	   -0.5
+        #   #   f	   -0.5
+        def percent_change periods = 1
+          value = []
+          arr = []
+          new_index = [] 
+          self.index.each do |e|
+            if self[e].is_a?(Numeric) 
```

Comment:
And btw, percent change for any data apart from numeric should not be supported.